parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveStringifier.java

```diff
@@ -331,6 +330,32 @@ private long convert(long duration, TimeUnit from, TimeUnit to, TimeUnit higher)
     }
   };
 
+  static PrimitiveStringifier createTimestampStringifier(final LogicalTypeAnnotation.TimeUnit timeUnit, final boolean isAdjustedToUTC) {
+    final TimeZone timeZone = isAdjustedToUTC ? TimeZone.getDefault() : TimeZone.getTimeZone("utc");
+    if (timeUnit == LogicalTypeAnnotation.TimeUnit.MILLIS) {
+      return new DateStringifier(
+        "TIMESTAMP_MILLIS_STRINGIFIER", "yyyy-MM-dd'T'HH:mm:ss.SSS z", timeZone) {
+        @Override
+        long toMillis(long value) {
+          return value;
+        }
+      };
+    }
+    return new DateStringifier(
+      "TIMESTAMP_MICROS_STRINGIFIER", "yyyy-MM-dd'T'HH:mm:ss.SSS z", timeZone) {
+      @Override
+      public String stringify(long value) {
+        return super.stringify(value) + String.format("%03d", Math.abs(value % 1000));
```

Comment:
Does this work as intended? It seems to me that it appends the micros after the already stringified millis-timestamp, but that does not end in the millis, but in the time zone instead (yyyy-MM-dd'T'HH:mm:ss.SSS z).

I would expect this to print something like "2018-08-17T07:41:12.345 America/Los_Angeles678".